tools/go-ls-tags/timeout/deadline.go

```diff
@@ -0,0 +1,21 @@
+package timeout
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+var ErrTimeoutExceeded = errors.New("timeout exceeded")
+
+func WithDeadline(ctx context.Context, fn func() error) error {
+	deadline, ok := ctx.Deadline()
```

Comment:
Perhaps the idiomatic way to work with deadlines would be something like this (you could place it directly in reader.go and get rid of this deadline.go file:
```
func scanFile(ctx context.Context, f *os.File, fn func(line string) error) error {
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		select {
		case <-ctx.Done():
			return ctx.Err()
		default:
			return fn(scanner.Text())
		}
	}

	if err := scanner.Err(); err != nil {
		return err
	}
	return nil
}
```
If the deadline is exceeded the `ctx.Err()` will return `DeadlineExceeded`

Reply:
Fixed in 7d95365e